en/performance/practical-search-performance-guide.md

```diff
@@ -0,0 +1,2278 @@
+      rank: filter
+      match: word
+    }
+    field title type string {
+      indexing: summary | index
+      index: enable-bm25
+    }
+    field artist type string {
+      indexing: summary | index
+    }
+    field tags type weightedset&lt;string&gt; {
+      indexing: summary | attribute
+      attribute: fast-search
+    }
+    field similar type tensor&lt;float&gt;(trackid{}) {
+      indexing: summary | attribute
+      attribute: fast-search 
+    }
+  }
+
+  fieldset default {
+    fields: title, artist
+  }
+
+  document-summary track_id {
+    summary track_id type string { 
+      source: track_id
+    }
+  }
+  rank-profile personalized {
+    first-phase {
+      expression: rawScore(tags)
+    }
+  }
+   rank-profile similar {
+    first-phase {
+      expression: sum(attribute(similar) * query(user_liked))
+    }
+  }
+}
+</pre>
+
+Deploy the application again :
+
+<div class="pre-parent">
+  <button class="d-icon d-duplicate pre-copy-button" onclick="copyPreContent(this)"></button>
+<pre data-test="exec">
+$ vespa deploy --wait 300 app
+</pre>
+</div>
+
+And again, since we added `fast-search` we need to restart the searchnode process:
+
+<div class="pre-parent">
+  <button class="d-icon d-duplicate pre-copy-button" onclick="copyPreContent(this)"></button>
+<pre data-test="exec">
+$ docker exec vespa bash -c "/opt/vespa/bin/vespa-sentinel-cmd restart searchnode"
+</pre>
+</div>
+
+Wait for the searchnode to start up 
+<pre>
+curl -s http://localhost:19110/state/v1/health
+</pre>
+
+<pre style="display:none" data-test="exec">
+$ sleep 60
+</pre>
+
+Let us re-run our query example :
+
+<div class="pre-parent">
+  <button class="d-icon d-duplicate pre-copy-button" onclick="copyPreContent(this)"></button>
+<pre data-test="exec" data-test-assert-contains="Bonnie Tyler">
+{% raw %}
+$ vespa query 'yql=select title,artist, track_id from track where !weightedSet(track_id,@userLiked)' \
+'ranking.features.query(user_liked)={{trackid:TRQIQMT128E0791D9C}:1.0,{trackid:TRWJIPT128E0791D99}:1.0,{trackid:TRGVORX128F4291DF1}:1.0}' \
+'ranking=similar' \
+'hits=5' \
+'userLiked={TRQIQMT128E0791D9C:1,TRWJIPT128E0791D99:1,TRGVORX128F4291DF1:1}'
+{% endraw %}
+</pre>
+</div>
+
+The `querytime` dropped down to 40 ms instead of 120 ms without the `fast-search` option. 
+See also [performance considerations](../tensor-user-guide.html#performance-considerations)
+when using tensor expressions and especially the tensor cell precision used.
+Vespa supports both `int8`, `bfloat16`, `float` and
+`double` precision cell types.  
+
+## Multi-threaded search and ranking 
+So far in this guide all our searches and ranking computations have been performed using 
+single threaded matching and ranking. To enable multi-threaded execution we 
+need to change our `services.xml`. 
+Multi-threaded search and ranking can improve query latency significantly and make
+use of multi-cpu core architectures better. Search and ranking has evolved since 1998 and by 
```

Comment:
```suggestion
Multi-threaded search and ranking can improve query latency significantly and make better
use of multi-cpu core architectures. Search and ranking has evolved since 1998 and by 
```